tests/framework/matcher/restarted.go

```diff
@@ -0,0 +1,55 @@
+package matcher
+
+import (
+	"fmt"
+
+	"github.com/onsi/gomega/types"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	k8stypes "k8s.io/apimachinery/pkg/types"
+
+	v1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/tests/framework/matcher/helper"
+)
+
+func BeRestarted(oldUID k8stypes.UID) types.GomegaMatcher {
+	return restartedMatcher{
+		oldUID: oldUID,
+	}
+}
+
+type restartedMatcher struct {
+	oldUID k8stypes.UID
+}
+
+func (r restartedMatcher) Match(actual interface{}) (success bool, err error) {
+	u, err := helper.ToUnstructured(actual)
+	if err != nil {
+		return false, nil
+	}
+	uid, _, err := unstructured.NestedString(u.Object, "metadata", "uid")
+	if err != nil {
+		return false, nil
+	}
+	phase, _, err := unstructured.NestedString(u.Object, "status", "phase")
+	if err != nil {
+		return false, nil
+	}
+	return uid != string(r.oldUID) && phase == string(v1.Running), nil
+}
+
+func (r restartedMatcher) FailureMessage(actual interface{}) (message string) {
+	u, err := helper.ToUnstructured(actual)
+	if err != nil {
+		return err.Error()
+	}
+	return fmt.Sprintf("Expected VMI to be restarted, but got '%v'", u)
```

Comment:
Can you use `format.Message()` as is used in other gomega matchers?

Reply:
Sure. You can take a look at implementation of any matcher in gomega. For example this is `Equal()`:

https://github.com/kubevirt/kubevirt/blob/25a4b67be739c9634898597ee958ee10ec718529/vendor/github.com/onsi/gomega/matchers/equal_matcher.go#L30-L38

And here is the `format.Message()` definition: 

https://github.com/kubevirt/kubevirt/blob/2f2b42e0d90a181a8fd5bc39aa43b194c895014f/vendor/github.com/onsi/gomega/format/format.go#L68-L87

Reply:
But note that the string representation of `actual` will be trunctuated to a maximum length `fromat.MaxLength`, which is 4000 by default.